20240129-satellite-managed-encryption.md

```diff
@@ -0,0 +1,124 @@
+
+Functionality should then be added to allow creating a project with a satellite managed encryption passphrase:
+* the new column in the `projects` table is set to `true` for this project
+* a new key-value pair is created for the new project in the Key Management Store. The passphrase should be cryptographically random
+* endpoints on the satellite API are added or updated so that an authenticated user can request the encryption passphrase for a project during access creation
+
+#### WebAssembly Updates
+
+Wasm functionality must be updated to allow the Satellite UI to properly generate access credentials with path encryption disabled. [Code for reference](https://github.com/storj/storj/blob/a62929fd5757a40eda5d0b044ad2cefc14708410/satellite/console/consolewasm/access.go#L22-L29).
+
+#### Satellite UI Updates
+
+When creating a project, the user should be provided the option to create different types of projects:
+
+* satellite-managed-encryption project
+* user-managed-encryption project
+
+The user-managed-encryption project will be the same as before, and user will always be prompted to enter encryption passphrase when necessary, e.g. when using the file browser or generating an access grant.
+
+The satellite-managed-encryption project will never prompt the user for an encryption passphrase in the UI. Instead, when the user does an action that requires generating credentials, the UI will retrieve the project's encryption passphrase from the satellite KMS, and generate an access grant using this passphrase, and path encryption disabled.
+
+The user should not be able to directly retrieve a "raw API key" from the Satellite UI. This prevents the user from unintentionally creating "invalid" credentials using a raw API key with Uplink. The only credentials that can be generated from the access page for a satellite-managed-encryption project should be credentials that already have embedded encryption rules.
+
+The file browser in the UI should generate access credentials using the passphrase from the satellite KMS, and no path encryption.
+
+#### Uplink Updates
+
+New versions of Uplink can be updated to check for the new flag on the project, and automatically create accesses with the correct rules during commands like `uplink setup` and `uplink access create`.
+
+This will allow new Uplink versions to properly handle "raw API keys" (access keys without information like satellite or encryption embedded) for "satellite-managed-encryption" projects.
+
+These changes are technically unnecessary if the UI is updated to not provide raw API keys for satellite-managed-encryption projects. However, it may be useful to make these changes to Uplink so that future versions of Uplink can use raw API keys even for satellite-managed-encryption projects.
```

Comment:
It is not really useful. We moved this to "out of scope" and most likely will not spend effort on this for the time being, since it is unnecessary with the other changes.